src/DotLiquid/ShopifyFilters.cs

```diff
@@ -0,0 +1,119 @@
+using System.Globalization;
+using System.Security.Cryptography;
+using System.Text;
+
+namespace DotLiquid
+{
+    /// <summary>
+    /// Shopify filters implemented by DotLiquid
+    /// </summary>
+    /// <see href="https://shopify.dev/docs/themes/liquid/reference/filters"/>
+    public static class ShopifyFilters
+    {
+        /// <summary>
+        /// Converts a string into an MD5 hash.
+        /// </summary>
+        /// <see href="https://shopify.dev/docs/themes/liquid/reference/filters/string-filters#md5"/>
+        /// <param name="input">The input to be encoded</param>
+        public static string Md5(string input)
+        {
+            if (input.IsNullOrWhiteSpace())
+                return input;
+
+            using (var hashAlgorithm = MD5.Create())
+            {
+                return ComputeHash(hashAlgorithm, input);
+            }
+        }
+
+        /// <summary> 
+        /// Converts a string into a SHA-1 hash.
+        /// </summary>
+        /// <see href="https://shopify.dev/docs/themes/liquid/reference/filters/string-filters#sha1" />
+        /// <param name="input">The input to be encoded</param>
+        public static string Sha1(string input)
+        {
+            if (input.IsNullOrWhiteSpace())
+                return input;
+
+            using (var hashAlgorithm = SHA1.Create())
+            {
+                return ComputeHash(hashAlgorithm, input);
+            }
+        }
+
+        /// <summary>
+        /// Converts a string into a SHA-256 hash.
+        /// </summary>
+        /// <see href="https://shopify.dev/docs/themes/liquid/reference/filters/string-filters#sha256" />
+        /// <param name="input">The input to be encoded</param>
+        public static string Sha256(string input)
+        {
+            if (input.IsNullOrWhiteSpace())
+                return input;
+
+            using (var hashAlgorithm = SHA256.Create())
+            {
+                return ComputeHash(hashAlgorithm, input);
+            }
+        }
+
+        /// <summary>
+        /// Converts a string into a SHA-1 hash using a hash message authentication code (HMAC).
+        /// Pass the secret key for the message as a parameter to the filter.
+        /// </summary>
+        /// <see href="https://shopify.dev/docs/themes/liquid/reference/filters/string-filters#hmac_sha1" />
+        /// <param name="input">The input to be encoded</param>
+        /// <param name="secretKey">The secret key</param>
+        public static string HmacSha1(string input, string secretKey)
+        {
+            if (input.IsNullOrWhiteSpace() || secretKey is null)
```

Comment:
```suggestion
            if (input == null || secretKey == null)
```
(just for consistency replaced is with ==)